### PR TITLE
[DX] Add more options descriptions to the docs

### PR DESCRIPTION
**Pull request by SpacePossum**
To keep the PR atomic I only modified the documentation this time around ;)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -516,18 +518,18 @@ only those exact fixers are enabled whether or not level is set.
 With the ``--config-file`` option you can specify the path to the
 ``.php_cs`` file.
 
-By using ``--using-cache`` option you can set if caching
-mechanism should be used.
+By using ``--using-cache`` option with ``yes`` or ``no`` you can set if caching
```
Comment (keradus) [inline README.rst]: this option is not enabled by default. It is not set by default therefore the configuration from `Config` class is taken.

**Comment by keradus** [comment]
Just one issue, everything other is great ;)

File: Symfony/CS/Console/Command/FixCommand.php [hunk 2]
```diff
@@ -113,12 +113,14 @@ protected function configure()
...
-problems as possible on a given file or directory:
+problems as possible on a given file or files in a given directory and its subdirectories:
 
     <info>php %command.full_name% /path/to/dir</info>
     <info>php %command.full_name% /path/to/file</info>
 
-The <comment>--verbose</comment> option show applied fixers. When using ``txt`` format (default one) it will also displays progress notification.
+The <comment>--format</comment> option can be used to set the output format of the results; ``txt`` (default one), ``xml`` or ``json``.
+
+The <comment>--verbose</comment> option will show the applied fixers. When using ``txt`` format it will also displays progress notifications.
```
Comment (sstok) [inline Symfony/CS/Console/Command/FixCommand.php]: when using **the** ``txt`` format

Or alternatively.

when using ``txt`` **as** format

File: Symfony/CS/Console/Command/FixCommand.php [hunk 3]
```diff
@@ -251,18 +253,18 @@ protected function configure()
 With the <comment>--config-file</comment> option you can specify the path to the
 <comment>.php_cs</comment> file.
 
-By using ``--using-cache`` option you can set if caching
-mechanism should be used.
+By using ``--using-cache`` option with ``yes`` or ``no`` you can set if caching
```
Comment (sstok) [inline Symfony/CS/Console/Command/FixCommand.php]: using **the** ``--using-cache`` ;)

File: README.rst (hunk 1 above)
Comment (SpacePossum) [inline README.rst]: I'm confused;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/README.rst#caching
``` The caching mechanism is enabled by default. ```
Is this on master only? (my PR targets master)

Comment (keradus) [inline README.rst]: Cache is enabled by default but that is not a case here.
Cache is enabled by default via Config class, not via console arguments.

File: README.rst [hunk 4]
```diff
@@ -4,7 +4,6 @@ PHP Coding Standards Fixer
 The PHP Coding Standards Fixer tool fixes *most* issues in your code when you
 want to follow the PHP coding standards as defined in the PSR-1 and PSR-2
 documents.
-
 If you are already using ``PHP_CodeSniffer`` to identify coding standards
```
Comment (keradus) [inline README.rst]: If I can as you - could you update it to be more general? PHPCS has it's own CS fixer now, then IMO it is worth to add here more general descr like `if you are using a linter to identify....` or sth

File: Symfony/CS/Console/Command/FixCommand.php [hunk 5]
```diff
@@ -242,27 +259,14 @@ protected function configure()
...
-Caching
--------
-
-The caching mechanism is enabled by default. This will speed up further runs by
-fixing only files that were modified. Tool will fix all files if tool version
-changed or fixers list changed.
-Cache is supported only for tool downloaded as phar file or installed via
-composer.
-Cache can be disabled via ``--using-cache`` option or config file:
+When using a configuration file the caching mechanism is enabled by default.
```
Comment (keradus) [inline Symfony/CS/Console/Command/FixCommand.php]: You have splitted info about cache.
When you described `--using-cache` option and talk how cache works, and few § later tell it is enabled by default.

Now, out of the blue, you are telling it is enabled by default.
One more hint - even if you don't have any `.php_cs` file then standard `Symfony\CS\Config\Config` will be used -  some instance of configuration is always used, and the behavior of cache set by config class can be overriden by option. Not only when one has a config file.

**Comment by keradus** [comment]
Why is this closed @SpacePossum ? change itself was great, just two minor questions from me ;)

**Comment by SpacePossum** [comment]
@keradus sorry for getting back so late about this PR. I closed it because I didn't have time to finish this any time soon (same reason I reply after so long).
I wonder if this PR will ever make it because I feel like I might not have enough indept knowledge about the Fixer to write good documentation. I happily take on the requested changes about the `files` vs `file` and the one about the `linter` and do a rebase.
However; I think you have a valid point about the proposed change of me to split the documention based on commandline arguments only and the configuration file, but I don't find a way to formulate a better solution.

File: README.rst [hunk 6]
```diff
@@ -437,13 +438,28 @@ fixed but without actually modifying them:
...
+By using the ``--using-cache`` option with ``yes`` or ``no`` you
+can set if the caching mechanism should be used. Using caching will speed up further
+runs by fixing only files that were modified since the last run. The tool will
+fix all files if the tool version has changed or the list of fixers has changed.
+The tool will generate ```.php_cs.cache``` files for caching.
+Caching is only supported when the tool is downloaded as phar file or installed via
+composer.
+
+Configuration file
+-------
```
Comment (stof) [inline README.rst]: wrong length for this line, making it invalid rST

Comment (SpacePossum) [inline README.rst]: Thanks will fix this.

**Pull request closed without merging**